server/lib/isErrnoException.ts

```diff
@@ -0,0 +1,12 @@
+import { isArbitraryObject } from '../../lib/utils/isArbitraryObject';
+
+export const isErrnoException = (error: unknown): error is NodeJS.ErrnoException => {
+	return (
+		isArbitraryObject(error) &&
+		error instanceof Error &&
+		(typeof error.errno === 'number' || typeof error.errno === 'undefined') &&
+		(typeof error.code === 'string' || typeof error.code === 'undefined') &&
+		(typeof error.path === 'string' || typeof error.path === 'undefined') &&
+		(typeof error.syscall === 'string' || typeof error.syscall === 'undefined')
+	);
+};
```

Comment:
This will also return true for any Error instance that doesn't have those 4 attributes.

Reply:
It's fine. The definition of `NodeJS.ErrnoException` ensures all these fields are optional. It's better to think about what is **NOT** a `NodeJS.ErrnoException`: something that is not an instance of `Error` and/or has one of the attributes with a _different_ type. For instance:

```ts
const valid = new Error();
isErrnoException(valid); // true

const invalid = Object.assign(new Error(), { errno: 'should be a number here' });
isErrnoException(invalid); // false
```

In that context, `isErrnoException` checks if the object's shape is compliant. Something like you are suggesting may be called `isStrictErrnoException`.